manticore/ethereum/manticore.py

```diff
@@ -572,6 +579,76 @@ def make_symbolic_arguments(self, types):
         # FIXME this is more naive than reasonable.
         return ABI.deserialize(types, self.make_symbolic_buffer(32, name='INITARGS', avoid_collisions=True))
 
+    def json_create_contract(self, jfile, owner=None, name=None, contract_name=None, balance=0, gas=None, network_id=None):
+        """ Creates a solidity contract based on a truffle json artifact.
+
+            :param jfile: truffle json artifact
+            :type jfile: str or IOBase
+            :param owner: owner account (will be default caller in any transactions)
+            :type owner: int or EVMAccount
+            :param contract_name: Name of the contract to analyze (optional if there is a single one in the source code)
+            :type contract_name: str
+            :param balance: balance to be transferred on creation
+            :type balance: int or BitVecVariable
+            :param gas: gas budget for each contract creation needed (may be more than one if several related contracts defined in the solidity source)
+            :type gas: int
+            :param network_id: Truffle network id to instantiate
+            :rtype: EVMAccount
+        """
+
+        if isinstance(jfile, io.IOBase):
+            jfile = jfile.read()
+        elif not isinstance(jfile, str):
+            raise TypeError(f'source code bad type: {type(jfile).__name__}')
+
+        truffle = json.loads(jfile)
+        hashes = {}
+        for item in truffle['abi']:
+            type = item['type']
+            if type in ('function'):
```

Comment:
See [#r263540882](https://github.com/trailofbits/manticore/pull/1376/files#r263540882)
```suggestion
            if item_type in ('function'):
```